dbms/src/Storages/StorageFile.cpp

```diff
@@ -223,7 +225,8 @@ class StorageFileBlockInputStream : public IBlockInputStream
         else
         {
             shared_lock = std::shared_lock(storage->rwlock);
-            read_buf = wrapReadBufferWithCompressionMethod(std::make_unique<ReadBufferFromFile>(file_path), compression_method);
+            file_path = std::make_optional(file_path_);
```

Comment:
Why std::optional here?

Reply:
Got it.

Reply:
Because of logic - there is only one case of three, when we have file_path.